pkg/csi/service/wcp/controller.go

```diff
@@ -512,22 +518,25 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
 					"failed to find shared datastores for given topology requirement. Error: %v", err)
 			}
+		} else if hostnameLabelPresent && isVdppOnStretchedSVEnabled {
+			log.Infof("Host Local volume provisioning with requirement: %+v", topologyRequirement)
 		} else {
-			if hostnameLabelPresent && isVdppOnStretchedSVEnabled {
-				log.Infof("Host Local volume provisioning with requirement: %+v", topologyRequirement)
-			} else {
-				// zone labels not Present in the topologyRequirement
-				if len(clusterComputeResourceMoIds) > 1 {
-					return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.FailedPrecondition,
-						"stretched supervisor cluster does not support creating volumes "+
-							"without zone keys in the topologyRequirement  . Error: %v", err)
-				}
-				sharedDatastores, vsanDirectDatastores, err = getCandidateDatastores(ctx, vc,
-					clusterComputeResourceMoIds[0], true)
-				if err != nil {
-					return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
-						"failed finding candidate datastores to place volume. Error: %v", err)
-				}
+			// No topology labels present in the topologyRequirement
+			if workloadDomainIsolationEnabled {
```

Comment:
if isVdppOnStretchedSV is disabled but workloadDomainIsolation is enabled, we don't want to fail createvolume call

Reply:
Updated it